mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -52,6 +52,10 @@ private[feature] trait StringIndexerBase extends Params with HasInputCol with Ha
     val outputFields = inputFields :+ attr.toStructField()
     StructType(outputFields)
   }
+
+  /** @group setParam */
+  def setHandleInvalid(value: String): this.type = set(handleInvalid, value)
```

Comment:
This may break Java compatibility (setters in traits don't seem to work).  I'd revert the change.